### PR TITLE
[Chore] Registers the TreeStore onto the bus

### PR DESCRIPTION
**Pull request by dylanlott**
## Description

tl;dr- a one-liner that registers the TreeStore submodule to the bus.

This was enabled by a change made in #855 after the TreeStore was modularized that creates a concrete Submodule implementation and allows them to be passed to the bus registration function like full modules. This PR updates the TreeStore to take full advantage of that submodule interface change and expose it on the bus for use in external modules, e.g. the IBC module.

## Issue

References a discussion abodut submodules and bus registration brought up during #808 
Updates the TreeStore submodule to register itself on the bus as a submodule which was enabled by #855 

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Registers the TreeStore submodule on the main bus

## Testing

- [x] `make develop_test`; if any code changes were made
- [x] `make test_e2e` on [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any code changes were made
- [ ] `e2e-devnet-test` passes tests on [DevNet](https://pocketnetwork.notion.site/How-to-DevNet-ff1598f27efe44c09f34e2aa0051f0dd); if any code was changed
- [ ] [Docker Compose LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md); if any major functionality was changed or introduced
- [ ] [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any infrastructure or configuration changes were made

## Required Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added, or updated, [`godoc` format comments](https://go.dev/blog/godoc) on touched members (see: [tip.golang.org/doc/comment](https://tip.golang.org/doc/comment))
- [ ] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

**Review by Olshansk — APPROVED** [review]

**Review by h5law — CHANGES_REQUESTED** [review]
@dylanlott Could you also add a function to the `bus` like so `GetTreeStore()` in `shared/modules/bus_module.go` and `runtime/bus.go` so the module can be retirved from the registry?

Also not sure if the treestore module has it but just incase the module registry requires the `GetModuleName()` function in order to register the module

**Comment by dylanlott** [comment]
Superseded by
